### PR TITLE
Add output for poetry lock --check

### PR DESCRIPTION
**Pull request by artemrys**
# Pull Request Check List

Resolves: #5038

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

**Comment by radoering** [comment]
I don't like the wording _"up to date"_ resp. _"out of date"_ because it is kind of ambiguous. The command does not check if some dependencies in the lock file are not up to date (as long as they comply with the constraints from `pyproject.toml`). The command only checks if `poetry.lock` is consistent with `pyproject.toml` (see [documentation](https://python-poetry.org/docs/master/cli/#options-9)). Maybe, we should stick with the wording from the documentation?

Further, I would say the minimum fix is running `poetry lock --no-update` (only making `poetry.lock` consistent with `pyproject.toml` not updating any dependencies already locked). Thus, I would propose to run `poetry lock [--no-update]` (square brackets for not required but sensible option).

I am aware that the warning when running `poetry install` says: `Warning: The lock file is not up to date with the latest changes in pyproject.toml. You may be getting outdated dependencies. Run update to update them.` I think this message is ok, because it defines _up to date_ more precisely by adding _with the latest changes in pyproject.toml_. However, I think `poetry lock [--no-update]` would be a better recommendation in this case, too.

All in all, I think the wording should be harmonized. Either **consistent with `pyproject.toml`** or **up to date with the latest changes in `pyproject.toml`**. Further, I would propose to recommend running `poetry lock [--no-update]` instead of `poetry update`.

But that is only my opinion. Any strong opinions from @python-poetry/triage ?

File: src/poetry/console/commands/lock.py [hunk 1]
```diff
@@ -37,11 +37,13 @@ def handle(self) -> int:
...
         if self.option("check"):
-            return (
-                0
-                if self.poetry.locker.is_locked() and self.poetry.locker.is_fresh()
-                else 1
+            if self.poetry.locker.is_locked() and self.poetry.locker.is_fresh():
+                self.line("poetry.lock is up to date")
+                return 0
+            self.line(
+                "Error: poetry.lock is out of date. Run `poetry update` to fix it"
```
Comment (radoering) [inline src/poetry/console/commands/lock.py:44]: You should use `<error>...</error>` as in https://github.com/python-poetry/poetry/blob/28d05d002475f0cc55e7cc46ae01f9a3ce4e03a8/src/poetry/console/commands/check.py#L26
for coloring.

**Review by radoering — CHANGES_REQUESTED** [review]

File: src/poetry/console/commands/lock.py (hunk 1 above)
Comment (artemrys) [reply src/poetry/console/commands/lock.py:44]: Should we wait on an opinion from traiate team on this one?

Comment (radoering) [reply src/poetry/console/commands/lock.py:44]: I think adding `<error>...</error>` is safe.

Considering the wording: Since there are no replies yet, either nobody had time to think about it or has no strong opinion or is just fine with the proposal. 😉 If you want to speed up the process (and are not afraid from some more iterations of changes) we can push this PR until I will approve it. Before merging, a core member will look at the changes anyway and may (or may not) request further changes.

Thus, if you want to proceed, I propose to stick with the wording from the documentation (`poetry.lock` is (not) consistent with `pyproject.toml`) and recommend running `poetry lock [--no-update]`. Otherwise, you can also just wait some more days for additional opinions.

Comment (finswimmer) [reply src/poetry/console/commands/lock.py:44]: > I propose to stick with the wording from the documentation (poetry.lock is (not) consistent with pyproject.toml) and recommend running poetry lock [--no-update]


I like this one :+1:

Comment (artemrys) [reply src/poetry/console/commands/lock.py]: Updated messages :)

**Comment by radoering** [comment]
Can you adapt the warnings in https://github.com/python-poetry/poetry/blob/08ed8ed14f8b4d079c5925156fbaa362b48e8cd6/src/poetry/installation/installer.py#L261-L264 and https://github.com/python-poetry/poetry/blob/08ed8ed14f8b4d079c5925156fbaa362b48e8cd6/src/poetry/console/commands/export.py#L62-L65 to the new wording, please?

I'd say we replace the first and the last sentence with the sentences from the error message and replace "outdated" in the second sentence by "improper" (or similar).

File: src/poetry/console/commands/lock.py [hunk 2]
```diff
@@ -37,11 +37,16 @@ def handle(self) -> int:
         )
 
         if self.option("check"):
-            return (
-                0
-                if self.poetry.locker.is_locked() and self.poetry.locker.is_fresh()
-                else 1
+            if self.poetry.locker.is_locked() and self.poetry.locker.is_fresh():
+                self.line("poetry.lock is consistent with pyproject.toml")
```
Comment (radoering) [inline src/poetry/console/commands/lock.py:41]: I would add a punctuation mark at the end of the message.

Comment (artemrys) [reply src/poetry/console/commands/lock.py]: Done

**Comment by artemrys** [comment]
I've updated `installer`'s and `export`'s messages to use new wording. I can add tests for that code in the next PR.

**Comment by radoering** [comment]
Sorry, for nitpicking again, but hopefully the last request for change from me. 😉 I didn't meant to change the warnings in `installer` and `export` to errors but only to change the wording. Due to the fact that the commands are not aborted, a warning seems to be more suitable than an error for these commands.

Just to be clear: The error for `poetry check --lock` is suitable.

File: src/poetry/console/commands/export.py [hunk 3]
```diff
@@ -57,12 +57,11 @@ def handle(self) -> None:
             self.call("lock", " ".join(options))
 
         if not locker.is_fresh():
-            self.line_error(
+            self._io.write_line(
```
Comment (radoering) [inline src/poetry/console/commands/export.py:60]: Probably, `line_error()` is correct here, because warnings should go to stderr instead of stdout. I know, in `installer.py` the warning is written to stdout... Maybe, something that can be harmonized for all warnings in a separate PR...

File: src/poetry/console/commands/export.py [hunk 4]
```diff
@@ -57,12 +57,11 @@ def handle(self) -> None:
...
                 "<warning>"
-                "Warning: The lock file is not up to date with "
-                "the latest changes in pyproject.toml. "
-                "You may be getting outdated dependencies. "
-                "Run update to update them."
+                "Error: poetry.lock is not consistent with pyproject.toml. "
```
Comment (radoering) [inline src/poetry/console/commands/export.py:62]: This should start with `"Warning: "`

File: src/poetry/installation/installer.py [hunk 5]
```diff
@@ -258,10 +258,9 @@ def _do_install(self, local_repo: Repository) -> int:
             if not self._locker.is_fresh():
                 self._io.write_line(
                     "<warning>"
-                    "Warning: The lock file is not up to date with "
-                    "the latest changes in pyproject.toml. "
-                    "You may be getting outdated dependencies. "
-                    "Run update to update them."
+                    "Error: poetry.lock is not consistent with pyproject.toml. "
```
Comment (radoering) [inline src/poetry/installation/installer.py:261]: This should also start with `"Warning: "`

File: src/poetry/console/commands/export.py (hunk 3 above)
Comment (artemrys) [reply src/poetry/console/commands/export.py]: Updated

File: src/poetry/console/commands/export.py (hunk 4 above)
Comment (artemrys) [reply src/poetry/console/commands/export.py]: Updated

File: src/poetry/installation/installer.py (hunk 5 above)
Comment (artemrys) [reply src/poetry/installation/installer.py]: Updated

**Review by radoering — COMMENTED** [review]
Sorry, I missed this before.

File: src/poetry/console/commands/export.py [hunk 6]
```diff
@@ -59,10 +59,9 @@ def handle(self) -> None:
         if not locker.is_fresh():
             self.line_error(
                 "<warning>"
-                "Warning: The lock file is not up to date with "
-                "the latest changes in pyproject.toml. "
-                "You may be getting outdated dependencies. "
-                "Run update to update them."
+                "Warning: poetry.lock is not consistent with pyproject.toml. "
+                "You may be getting improper dependencies. "
+                "Run `poetry update` to fix it."
```
[the remark this answers is not in the document]
Comment (radoering) [reply src/poetry/console/commands/export.py:64]: Same in `installer.py`.

[the remark this answers is not in the document]
Comment (artemrys) [reply src/poetry/console/commands/export.py]: Done

**Review by radoering — APPROVED** [review]

**Comment by radoering** [comment]
Finally looks good to me.

@finswimmer : Can you approve running the workflows and consider merging if the tests pass?

**Review by finswimmer — APPROVED** [review]
